backend/OKPluginVariableRendering/VariableRendering.cs

```diff
@@ -0,0 +1,379 @@
+﻿using Microsoft.Extensions.Logging;
+using Newtonsoft.Json.Linq;
+using Omnikeeper.Base.CLB;
+using Omnikeeper.Base.Entity;
+using Omnikeeper.Base.Entity.DataOrigin;
+using Omnikeeper.Base.Model;
+using Omnikeeper.Base.Utils.ModelContext;
+using Omnikeeper.Entity.AttributeValues;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using System.Text.RegularExpressions;
+
+namespace OKPluginVariableRendering
+{
+    public class VariableRendering : CLBBase
```

Comment:
This CLB only needs to run when there are actual changes in the dependent layer(s). Otherwise, the run can be skipped. The framework for this is already implemented. The thing you need to do to support this in your CLB here is implement/override the method `protected override ISet<string> GetDependentLayerIDs(JObject config, ILogger logger)`. It should return the list of dependent layer IDs. In your case, it should return what's in `Configuration.InputLayerSet` as that are the layers this CLB depends on.

For an example implementation of that method, see https://github.com/max-bytes/omnikeeper/blob/27d038b00fd94eab44a12bd13190d90a01d53a51/backend/OKPluginCLBDummy/CLBDummy.cs#L36